airflow/jobs.py

```diff
@@ -2042,15 +2042,6 @@ def _execute(self):
         try:
             self.task_runner.start()
 
-            ti = self.task_instance
```

Comment:
Interesting, this place did not contain proper locking and refresh of the data: The process could have had outdated to work with and could have also been changing data while it was working with outdated data.